src/js/button.js

```diff
@@ -7,13 +12,13 @@
  * @class
  * @constructor
  */
-vjs.Button = vjs.Component.extend({
+var Button = Component.extend({
   /**
    * @constructor
    * @inheritDoc
    */
   init: function(player, options){
```

Comment:
I'd love to pull the plug and just move forward. Here's a few things I'm concerned about.
- Even if we make it really easy to migrate it still requires action from the plugin authors, and that could take a long time if it happens at all. Enough API breakages and our plugin ecosystem starts over from scratch with 5.0.
- 4.0 versions of the player are going to be around for a while for various reasons (3.0 is still around). Should plugin authors immediately break their support for those players, or find some way to support both on their own?

It'd probably be worth doing a review of the plugins and seeing which ones use this. All of the techs do at least, but that's not a ton.